launcher/FileSystem.cpp

```diff
@@ -872,16 +831,35 @@ QString RemoveInvalidPathChars(QString path, QChar replaceWith)
     invalidChars = BAD_WIN_CHARS;
 #endif
 
-    QString fileSystemType = getFileSystemType(QFileInfo(path).absolutePath());
-
-    if (fileSystemType == "FAT32") {
-        invalidChars += BAD_FAT32_CHARS;
-    } else if (fileSystemType == "NTFS") {
-        invalidChars += BAD_NTFS_CHARS;
-    } else if (fileSystemType == "HFS+" || fileSystemType == "APFS") {
-        invalidChars += BAD_HFS_CHARS;
-    } else if (fileSystemType == "exFAT") {
-        invalidChars += BAD_EXFAT_CHARS;
+    switch (statFS(path).fsType) {
+        case FilesystemType::FAT:
+            invalidChars += BAD_FAT_CHARS;
+            break;
+        case FilesystemType::NTFS:
+            invalidChars += BAD_NTFS_CHARS;
+            break;
+        // case FilesystemType::REFS:
```

Comment:
REFS afaik has the same limitations in names as NTFS but considering it's covered it only availble on window it should be covered the default bad win chars above. still feell it should be added to the NTFS fall-through